medusa/reconstruct/expand.py

```diff
@@ -285,137 +97,68 @@ def _continuous_iterative_binary_gapfill(model,phenotype_dict,cycle_order,
     variables = chain(*reaction_variables)
     for variable in variables:
         coefficients[variable] = 0.0
+    # update the linear coefficients for the added reaction to the penalitties of the reaction
+    new_coefficients =  {}
+    for reaction in [rxn.id for rxn in gapfiller.reactions]:
+        if reaction in reaction_probability.keys():
+            new_coefficients[reaction] = 1-reaction_probability[reaction]
+    for react_id in new_coefficients.keys():
+        reaction = gapfiller.reactions.get_by_id(react_id)
+        for_var = reaction.forward_variable
+        rev_var = reaction.reverse_variable
+        if coefficients[for_var]>0.0:
+            coefficients[for_var] = new_coefficients[react_id]
+        if coefficients[rev_var]>0.0:
+            coefficients[rev_var] = new_coefficients[react_id]
```

Comment:
This section looks great, but will need to be migrated over to the gapfill_to_ensemble function